application/src/main/java/org/opentripplanner/routing/linking/VertexLinker.java

```diff
@@ -743,15 +743,18 @@ private void createEdges(
     double length = SphericalDistanceLibrary.distance(to.getCoordinate(), from.getCoordinate());
     // apply consistent NoThru restrictions
     // if all joining edges are nothru, then the new edge should be as well
+    // 'from' is the new vertex to be connected, so check the 'to' vertex connections
     var incomingNoThruModes = getNoThruModes(to.getIncoming());
-    var outgoingNoThruModes = getNoThruModes(to.getIncoming());
+    var outgoingNoThruModes = getNoThruModes(to.getOutgoing());
     AreaEdgeBuilder areaEdgeBuilder = new AreaEdgeBuilder()
       .withFromVertex(from)
       .withToVertex(to)
       .withGeometry(line)
       .withName(hit.getName())
       .withMeterLength(length)
       .withPermission(hit.getPermission())
+      .withBicycleSafetyFactor((float) hit.getBicycleSafety())
```

Comment:
Would it not make sense to convert the safety factors in `Area` to float? I can't see a use case for using a double there but a float elsewhere.

Reply:
I agree that the conversion should be done when it is retrieved from the `WayProperties` and stored in the `Area`. It has to be double in `WayProperties` because the mixin calculation is done there so the extra precision is needed.